src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs

```diff
@@ -106,6 +108,46 @@ private IEnumerable<EndpointConfig> ReadEndpoints()
             return endpoints;
         }
 
+        private Dictionary<string, SniConfig> ReadSni(IConfigurationSection sniConfig)
+        {
+            var sniDictionary = new Dictionary<string, SniConfig>(0, StringComparer.OrdinalIgnoreCase);
+
+            foreach (var sniChild in sniConfig.GetChildren())
+            {
+                // "SNI": {
+                //     "a.example.org": {
+                //         "Protocols": "Http1AndHttp2",
```

Comment:
This protocol is used for ALPN. How does it interact with the protocol configured on the endpoint? Do we need to flow this value up in the handshake results feature so we know what was requested vs what was negotiated?